test/utils/loader.cpp

```diff
@@ -0,0 +1,154 @@
+                        { "row", int(round(row)) },
+                        { "col", int(round(col)) }
+                    });
+                    count++;
+                }
+            }
+            
+            if (parent_label == "") this->parsed_[label] = res;
+            else this->parsed_[parent_label][label] = res;
+        }
+
+    public:
+        Parser(const std::string& path, const int& test_id): file_path_{ path } {
+            this->parsed_["test_id"] = test_id;
+        }
+
+        void parseFile(const int& rows, const int& cols, const int& coord_range) {
+            std::ifstream file(this->file_path_);
+            while (std::getline(file, this->line_, '\r')) {
+                // input
+                if (this->line_.find("MATRIX") != std::string::npos) {
+                    this->parseMatrix(file, "", "input", rows, cols);  
+                }
+
+                // energy
+                else if (this->line_.find("SOBEL") != std::string::npos) {
+                    this->parseMatrix(file, "energy", "sobel", rows, cols);
+                } else if (this->line_.find("VERT MAP") != std::string::npos) {
+                    this->parseMatrix(file, "energy", "map - vertical", rows, cols);
+                } else if (this->line_.find("HORZ MAP") != std::string::npos) {
+                    this->parseMatrix(file, "energy", "map - horizontal", rows, cols);
+                }
+
+                // seams
+                else if (this->line_.find("VERT SEAM") != std::string::npos) {
+                    this->parseSeam(file, "carver", "find - vertical", cols, coord_range);
+                } else if (this->line_.find("HORZ SEAM") != std::string::npos) {
+                    this->parseSeam(file, "carver", "find - horizontal", rows, coord_range);
+                }
+
+                // removes
+                else if (this->line_.find("REMOVE VERT") != std::string::npos) {
+                    this->parseMatrix(file, "carver", "remove - vertical", rows, cols - 1);
+                } else if (this->line_.find("REMOVE HORZ") != std::string::npos) {
+                    this->parseMatrix(file, "carver", "remove - horizontal", rows - 1, cols);
+                }
+                
+                // inserts
+                else if (this->line_.find("INSERT VERT") != std::string::npos) {
+                    this->parseMatrix(file, "carver", "insert - vertical", rows, cols + 1);
+                } else if (this->line_.find("INSERT HORZ") != std::string::npos) {
+                    this->parseMatrix(file, "carver", "insert - horizontal", rows + 1, cols);
+                }
+            }
+
+            file.close();
+        }
+
+        const json& getJSON() {
+            return this->parsed_;
+        };
+};
+
+int main() {
+    // count the number of seam carving tests
+    int test_count = 0;
+    std::string test_data_dir = "test/raw-data/";
+    for (const auto& entry : fs::directory_iterator(test_data_dir)) {
```

Comment:
`entry.path()` would've given me the direct path to the file, but it wouldn't have given me them in numerical order so I decided to do it myself.